### PR TITLE
EOS-24393 : [Containerization] Implementation of Hare miniprovisioner in k8s environment

### PR DESCRIPTION
**Pull request by suvratjoshi**
Implementation of key changes to sync with new confstore key changes.

Signed-off-by: Suvrat Joshi <suvrat.joshi@seagate.com>

### CODE REVIEW

**Comment by mssawant** [comment]
@suvratjoshi, did all the min provisioner unit tests pass for you? we can try to generate the required conf-store and invokr Hare mini-provisioner stages t o confirm if everything is working fine.

**Comment by mssawant** [comment]
@suvratjoshi @SwapnilGaonkar7 , what kind of testing is done on this PR?

File: provisioning/miniprov/hare_mp/main.py [hunk 1]
```diff
@@ -129,9 +136,17 @@ def logrotate(url: str):
...
-                '/etc/logrotate.d/hare')
+            with open(f'/opt/seagate/cortx/hare/conf/logrotate/{server_type}',
+                      'r') as f:
+                content = f.read()
+
+            log_dir = get_log_dir(url)
+            content = content.replace('/var/log/seagate/hare/*.log',
+                                      f'{log_dir}*.log')
+
+            with open('/etc/logrotate.d/hare', 'w') as f:
```
Comment (mssawant) [inline provisioning/miniprov/hare_mp/main.py:147]: Use, $config_dir before `/etc`.

File: utils/update-consul-conf [hunk 2]
```diff
@@ -339,4 +339,4 @@ sudo cp $CONF_FILE /etc/consul.d/
 sudo sed -r "s;(http://)localhost;\1$(get_node_name);" \
          -i $CONF_FILE
 
-consul reload > /dev/null
+#consul reload > /dev/null
```
Comment (mssawant) [inline utils/update-consul-conf:342]: We need this for LC.

**Review by mssawant — APPROVED** [review]
Looks good, minor comments. No further rounds are required. Just make sure all the tests are passed and post_install, prepare and config stages of mini provisioner are working fine.

**Review by SwapnilGaonkar7 — APPROVED** [review]

**Comment by suvratjoshi** [comment]
Testing done : 
```
[root@ssc-vm-6327 e24393]# /opt/seagate/cortx/hare/bin/hare_setup prepare --config yaml:///tmp/e24393/single_node_confstore.yaml

[root@ssc-vm-6327 e24393]# consul kv get --keys
epoch
eq-epoch
last_fidk
leader
m0conf/
ssc-vm-6327.colo.seagate.com/
[root@ssc-vm-6327 e24393]# consul kv get ssc-vm-6327.colo.seagate.com
Error! No key exists at: ssc-vm-6327.colo.seagate.com
[root@ssc-vm-6327 e24393]#
[root@ssc-vm-6327 e24393]# consul kv get --recurse ssc-vm-6327.colo.seagate.com
ssc-vm-6327.colo.seagate.com/dev/sdb:{"path": "/dev/sdb", "size": 26843545600, "blksize": 4096}
ssc-vm-6327.colo.seagate.com/dev/sdc:{"path": "/dev/sdc", "size": 26843545600, "blksize": 4096}
ssc-vm-6327.colo.seagate.com/facts:{
  "memorysize_mb": 3787.1328125,
  "processorcount": 1
}
```

```
[root@ssc-vm-6327 e24393]# /opt/seagate/cortx/hare/bin/hare_setup config --config yaml:///tmp/e24393/single_node_confstore.yaml
[root@ssc-vm-6327 e24393]#

[root@ssc-vm-6327 e24393]# ls /etc/cortx/hare/config/85fffa84fee1b12b46bcfc7cc491e602/
cluster.yaml        confd.dhall         confd.xc            consul-agents.json  consul-client-conf/ consul-env          consul-kv.json      consul-server-conf/ node-name           sysconfig/
[root@ssc-vm-6327 e24393]# ls /etc/cortx/hare/config/85fffa84fee1b12b46bcfc7cc491e602/sysconfig/
motr/ s3/
[root@ssc-vm-6327 e24393]# ls /etc/cortx/hare/config/85fffa84fee1b12b46bcfc7cc491e602/sysconfig/motr/ssc-vm-6327.colo.seagate.com/m0d-0x7200000000000001\:0x
m0d-0x7200000000000001:0x16  m0d-0x7200000000000001:0x9

[root@ssc-vm-6327 e24393]# ls /etc/cortx/hare/config/85fffa84fee1b12b46bcfc7cc491e602/sysconfig/s3/ssc-vm-6327.colo.seagate.com/s3server-0x7200000000000001\:0x
s3server-0x7200000000000001:0x19  s3server-0x7200000000000001:0x1f  s3server-0x7200000000000001:0x25  s3server-0x7200000000000001:0x2b  s3server-0x7200000000000001:0x31  s3server-0x7200000000000001:0x37
s3server-0x7200000000000001:0x1c  s3server-0x7200000000000001:0x22  s3server-0x7200000000000001:0x28  s3server-0x7200000000000001:0x2e  s3server-0x7200000000000001:0x34

[root@ssc-vm-6327 e24393]# ls /etc/cortx/hare/config/85fffa84fee1b12b46bcfc7cc491e602/ -l
total 76
-rw-r--r-- 1 root root   929 Sep 30 16:15 cluster.yaml
-rw-r--r-- 1 root root 16620 Sep 30 16:15 confd.dhall
-rw-r--r-- 1 root root  6647 Sep 30 16:15 confd.xc
-rw-r--r-- 1 root root   150 Sep 30 16:15 consul-agents.json
drwxr-xr-x 2 root root  4096 Sep 30 16:15 consul-client-conf
-rw-r--r-- 1 root root   299 Sep 30 16:15 consul-env
-rw-r--r-- 1 root root 19938 Sep 30 16:15 consul-kv.json
drwxr-xr-x 2 root root  4096 Sep 30 16:15 consul-server-conf
-rw-r--r-- 1 root root    28 Sep 30 16:15 node-name
drwxr-xr-x 4 root root  4096 Sep 30 16:15 sysconfig
[root@ssc-vm-6327 e24393]#

[root@ssc-vm-6327 e24393]# date
Thu Sep 30 16:17:59 MDT 2021
```

```
[root@ssc-vm-6327 e24393]# cd /etc/cortx/hare/config/85fffa84fee1b12b46bcfc7cc491e602/

[root@ssc-vm-6327 85fffa84fee1b12b46bcfc7cc491e602]# vi cluster.yaml

[root@ssc-vm-6327 85fffa84fee1b12b46bcfc7cc491e602]# vi confd.xc
```

**Comment by suvratjoshi** [comment]
Confstore yaml file used for above shown testing : 
```
cluster:
  storage_set:
  - durability:
      sns:
        data: '1'
        parity: '0'
        spare: '0'
    name: storage-set-1
    nodes:
    - 85fffa84fee1b12b46bcfc7cc491e602
  storage_set_count: '1'
cortx:
  common:
    product_release: LC
    service:
      admin: admin
      secret: !!binary |
        Z0FBQUFBQmhTYVdTSFVlU2dwNEhmdi1WdVhpRVEyRmo3SFREX0d5RHNzcDVSZGNXSDhodjZzeDdl
        R2JRWDlzU1FMWDBuSGFOekNKLWxrTHMweUQ2WjJSaEgzUzlpWUZ3dFE9PQ==
    setup_type: K8
    storage:
      config_b: /etc/cortx
      local: /etc/cortx
      log: /share/var/log/cortx
      misc: /var/data/cortx
      shared: /share/var/data/cortx
  csm:
    agent:
      endpoints:
      - https://csm.seagate.com:8081
    auth_admin: admin
    auth_secret: !!binary |
      Z0FBQUFBQmhTYVdTZVpqWUhveGRGSERXcVVGWXI3SnFuZTZqX3Jrb0pMNXVMRHZLOHAtd3VPMjNM
      eWY0X01UVDl3WTA5M1pOdERzRVE3SENjMkUyWmFucFd6bVNETk1sUUE9PQ==
    email_address: cortx@seagate.com
    mgmt_admin: cortxadmin
    mgmt_secret: !!binary |
      Z0FBQUFBQmhTYVdTMmZmTTNzb0tsZzdxN1VFODFTaURDNzZkRkUyLTItZFFuZWw5bFNkcGMwc29j
      dEhPTUFpWnZodDIxd2hHME8tc1dqcFR5dVh4NnNfUlNyd2V4N3R5UVE9PQ==
  external:
    consul:
      admin: admin
      endpoints:
      - consul-server1.cortx-cluster.com
      secret: !!binary |
        Z0FBQUFBQmhTYVdTVnNkM2duQnRnR0t3a0Y4eWRWdFNYbk11TkpQYW5rS2J1UnNIUDl2d2VuUjRv
        VEtwODZ4TTZPTnQxUE42US1NMFBuUkdzMzBvVG5GSWZfYkVEZkRfb2c9PQ==
    kafka:
      endpoints:
      - kafka-server1.cortx-cluster.com
      - kafka-server2.cortx-cluster.com
      secret: !!binary |
        Z0FBQUFBQmhTYVdTSEhVNEhaYUdxQjlncnBvXzZLSVJoWnlOSks3UU45R3o2WUh0blJ3VmJ0b0Zo
        VDRGbm1WekVucWxVNVM1UzFiN0MzcHU4RUtReVdTTjZyZXhWZlFlVUE9PQ==
      user: admin
    openldap:
      admin: admin
      base_dn: dc=seagate,dc=com
      endpoints:
      - oldap-server1.cortx-cluster.com
      secret: !!binary |
        Z0FBQUFBQmhTYVdTdEtTT3ozOGZ6Z1hVamtlNDc4UzEwd3U3b1JJM2Vsd2JjZzBGNVdvZ1c4cWl5
        bFZaQVUzdjdILUFTSmtBeGt6VVhheXB2cGlCdzhnUjY3OE4wRTVqTlE9PQ==
      servers:
      - oldap-server1.cortx-cluster.com
      - oldap-server2.cortx-cluster.com
      - oldap-server3.cortx-cluster.com
  hare:
    ports:
      endpoints:
      - tcp://hax.seagate.com:28000
  motr:
    client_instances: 2
    interface_family: inet
    interface_type: tcp
    ios:
      endpoints:
      - motr://motr-data1.cortx-cluster.com:3232
    transport_type: libfab
  s3:
    data:
      endpoints:
      - http://s3.seagate.com:80
      - https://s3.seagate.com:443
    iam:
      endpoints:
      - https://s3-iam.seagate.com:9443
    internal:
      endpoints:
      - http://s3.seagate.com:28049
    io_max_units: 32
    service_instances: 11
  utils:
    message_bus_backend: kafka
node:
  85fffa84fee1b12b46bcfc7cc491e602:
    cluster_id: 3e765bbb-8425-4e7f-8ee3-487e65ff2c8c
    components:
    - name: motr
      services:
      - io
    - name: hare
    - name: s3
      services:
      - io
      - auth
      - bg_consumer
    - name: utils
      services:
      - message_bus
    hostname: ssc-vm-6327.colo.seagate.com
    name: podnode-1
    network:
      data:
        private_fqdn: ssc-vm-6327.colo.seagate.com
    storage:
      cvg:
      - devices:
          data:
          - /dev/sdb
          - /dev/sdc
          name: cvg-01
          type: cas
      cvg_count: 1
    storage_set: storage-set-1
    storage_set_count: 1
    storage_set_id: storage-set-1
    type: storage_node
```

**Pull request merged**